### PR TITLE
Refactor controler and validators

### PR DESCRIPTION
**Pull request by lufercc**
Please take a look at this merge request

### CODE REVIEW

File: src/com/jalasoft/search/common/Helper.java [hunk 1]
```diff
@@ -46,7 +48,8 @@ public int convertToBytes(int numToConvert, int unit){
      * @param number on string to convert on int
      * @return int number converted
      */
-    public int convertStringtoInt(String number) {
-        return Integer.valueOf(number);
+    public int convertStringToInt(String number) {
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Helper.java]: delete space

File: src/com/jalasoft/search/common/Validator.java [hunk 2]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
         }
         return is_correct;
     }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java:75]: Add comment

File: src/com/jalasoft/search/common/Validator.java [hunk 3]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+    public boolean isPathDirection(String pathText) {
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: review the method name

File: src/com/jalasoft/search/common/Validator.java [hunk 4]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+        File f = new File(pathText);
+        return f.exists();
+    }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: add comment

File: src/com/jalasoft/search/common/Validator.java [hunk 5]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+    public boolean isValidDate(Date date) {
+        if(date instanceof Date)
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: return date instanceOf Date

File: src/com/jalasoft/search/common/Validator.java [hunk 6]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+            return true;
+        return false;
+    }
+
+    public boolean dateFromIsLessThanTo(Date from, Date to) {
+        if(from.compareTo(to) <= 0){
+            return true;
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: maybe could be better a direct return

File: src/com/jalasoft/search/common/Validator.java [hunk 7]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+        File f = new File(pathText);
+        return f.exists();
+    }
+
+    public boolean isValidDate(Date date) {
+        if(date instanceof Date)
+            return true;
+        return false;
+    }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: add comment

File: src/com/jalasoft/search/common/Validator.java [hunk 8]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+    public boolean dateFromIsLessThanTo(Date from, Date to) {
+        if(from.compareTo(to) <= 0){
+            return true;
+        }
+        return false;
+    }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: add comment could be better a direct return

File: src/com/jalasoft/search/common/Validator.java [hunk 9]
```diff
@@ -66,4 +72,36 @@ public boolean isFileCorrect(String file) {
...
+    public boolean isNotEmpty(String num) {
+        if (!num.isEmpty()){
+            return true;
+        }
+        return false;
+    }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/common/Validator.java]: add comment could be better a direct return

File: src/com/jalasoft/search/controller/Controller.java [hunk 10]
```diff
@@ -13,12 +13,12 @@
 import com.jalasoft.search.common.Helper;
 import com.jalasoft.search.common.Validator;
 import com.jalasoft.search.model.Asset;
-import com.jalasoft.search.model.QueryManager;
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:16]: are you sure about delete this lines?

File: src/com/jalasoft/search/controller/Controller.java [hunk 11]
```diff
@@ -13,12 +13,12 @@
...
 import com.jalasoft.search.view.MainWindow;
-import com.jalasoft.search.model.FileSearch;
 import com.jalasoft.search.model.Search;
-
-import java.sql.SQLException;
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java]: are you sure about delete this lines

File: src/com/jalasoft/search/controller/Controller.java [hunk 12]
```diff
@@ -13,12 +13,12 @@
...
+import java.util.Date;
+import java.util.HashMap;
+import static com.jalasoft.search.common.Log.getInstance;
+import static com.jalasoft.search.common.Log.getInstance;
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java]: twice import?

File: src/com/jalasoft/search/controller/Controller.java [hunk 13]
```diff
@@ -31,6 +31,8 @@
     private Validator validator;
     private Boolean advancedFlag;
     private Helper helper;
+    private boolean advanced;
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java]: use advancedFalg

File: src/com/jalasoft/search/controller/Controller.java [hunk 14]
```diff
@@ -42,65 +44,193 @@ public Controller(MainWindow window, Search search){
...
+        }
+        return sCriteria;
+    }
+
+    /**
+     * Method to validate the Size grater than
+     * @param min, max: the min should be less than maz
+     * @return true if meet the criteria
+     */
+    private boolean validateSizeGraterThan(int min, int max) {
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:119]: the validation should be go inside Validator class

File: src/com/jalasoft/search/controller/Controller.java [hunk 15]
```diff
@@ -42,65 +44,193 @@ public Controller(MainWindow window, Search search){
...
+        }
+        return false;
+    }
+
+    /**
+     * Method to validate a Date is greater than other
+     * @param from, to: the from should be less than to date
+     * @return true if meet the criteria
+     */
+    private boolean validateDateGraterThan(Date from, Date to) {
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:134]: the validation should be go inside Validator class

File: src/com/jalasoft/search/controller/Controller.java [hunk 16]
```diff
@@ -42,65 +44,193 @@ public Controller(MainWindow window, Search search){
...
+                    getInstance().getLogger().error("The Date is not Setup correctly");
+                }
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Method to validate the String input is valid String
+     * @param text String to evaluate teh Criteria
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:150]: the

File: src/com/jalasoft/search/controller/Controller.java [hunk 17]
```diff
@@ -42,65 +44,193 @@ public Controller(MainWindow window, Search search){
...
+        }
+        return res;
+    }
+
+    /**
+     * Method to validate is the path is valid
+     * @param pathText validate the String path exist
+     * @return True if the path exist
+     */
+    private boolean validatePath(String pathText) {
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:173]: the validation should be go inside Validator class

File: src/com/jalasoft/search/controller/Controller.java [hunk 18]
```diff
@@ -42,65 +44,193 @@ public Controller(MainWindow window, Search search){
...
+     * Method to validate all files and set search criteria with valid values
+     */
 
+
+    /**
+     * Method to transform the String setup into Int also convert it in bites
+     * @param intValue String to Transform in Integer
+     * @return int value in bites
+     */
+    private int getIntValue(String intValue) {
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/Controller.java:210]: the validation should be go inside Validator class

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 19]
```diff
@@ -111,15 +111,15 @@ public void setExtension(String extension) {
...
+     * @param hidden, flag to search between hidden or not hidden
      */
     public void setHidden(int hidden) {
         isHidden = hidden;
     }
 
     /**
-     * Method to set if a file is hidden or not
-     * @param type, true if the file is hidden otherwise false
+     * Method to set type of file to serch flag
```
Comment (Anaeli) [inline src/com/jalasoft/search/controller/SearchCriteria.java:121]: search

File: src/com/jalasoft/search/model/Search.java [hunk 20]
```diff
@@ -313,28 +311,24 @@ public Search() {
      * listAllFilesInPath method
      * @param folder this Param is a File object with the path setup
      * @param res is an ArrayList of FileSearch what is filled the this method
-     * */
-    private void listFilesForFolder(File folder , ArrayList<Asset> res) {
+     * @return true if finished  */
```
Comment (Anaeli) [inline src/com/jalasoft/search/model/Search.java]: */ in other line

File: src/com/jalasoft/search/view/HeaderPanel.java [hunk 21]
```diff
@@ -20,4 +20,11 @@ public void setErrorMessage(String errorMessage){
     public void cleanErrorMessage(){
         this.errorMessage.setText("");
     }
+
```
Comment (Anaeli) [inline src/com/jalasoft/search/view/HeaderPanel.java:23]: add comment

File: src/com/jalasoft/search/controller/Controller.java (hunk 10 above)
Comment (lufercc) [reply src/com/jalasoft/search/controller/Controller.java:16]: this line was added by Rodrigo

File: src/com/jalasoft/search/controller/Controller.java (hunk 11 above)
Comment (lufercc) [reply src/com/jalasoft/search/controller/Controller.java]: same, line added by rodrigo

File: src/com/jalasoft/search/controller/Controller.java (hunk 14 above)
Comment (lufercc) [reply src/com/jalasoft/search/controller/Controller.java:119]: Validator  are in common I think is not the correct place to add it

File: src/com/jalasoft/search/controller/Controller.java (hunk 10 above)
Comment (lufercc) [reply src/com/jalasoft/search/controller/Controller.java:16]: I will add them

**Review by rcespedesm — APPROVED** [review]

**Pull request merged**